swe_roe/ex2.c

```diff
@@ -719,6 +719,8 @@ struct _n_RDyApp {
   char filename[PETSC_MAX_PATH_LEN];
   /// PETSc grid
   DM dm;
+  /// A DM for creating PETSc Vecs with 1 DOF
```

Comment:
Maybe it would be good to think of these two DMs in terms of primary/secondary or prognostic/diagnostic variables. It's true that the "primary" DM holds the grid, which is shared between the two DMs, but I think a DM is really a set of DOF and their related function spaces.

This is just an FYI--I don't think anything is needed here unless you wanted to add a comment or change how you're referring to the DMs.

Reply:
I agree that we should use better variable names for the RDycore lib, but current names are good enough for this repo.